lib/module.ts

```diff
@@ -8,35 +8,25 @@ import {
   IMutations,
   IPlugin,
   IState,
-  ISubModules,
 } from './interfaces'
 
+/* eslint-disable @typescript-eslint/no-explicit-any */
 export function useModule<
-  S extends IState,
-  G extends IGetters,
-  M extends IMutations,
-  SM extends ISubModules = never,
->(options: IModuleOptions<S, G, M, SM>): IModule<S, G, M, SM> {
-  return new Module<S, G, M, SM>(options)
-}
-
-export function useRootModule<SM extends ISubModules>(
-  options: IModuleOptions<IState, IGetters, IMutations, SM>,
-): IFlattenedModule<IState, IGetters, IMutations, SM> {
-  const module = new Module<IState, IGetters, IMutations, SM>(options)
+  S extends IState = any,
+  G extends IGetters = any,
+  M extends IMutations = any,
+>(options: IModuleOptions<S, G, M>): IFlattenedModule<S, G, M> {
```

Comment:
I have to add `= any` here so that users can just `useModule({ ... })` without explicitly set the types for state, getters, and mutations.
I'm not sure if it's a good idea to allow users to do this, if not we can just remove all of the `= any` here, that way we will be able to force users to `useModule<S, G, M>({ ... })`.

Reply:
I think it's better to require types for now, mainly because it's easier to remove the restriction later than it is to add it. Also, I can't really think of a situation where we'd want a module without typings.

Reply:
I see, no problem!
I'll change it now, thank you!

Reply:
I've removed the `= any` in the latest commit.
Please check it, thanks!

Reply:
Looks good 🙏